### ISSUE TITLE
[BUG] ARM code doesn't get interpreted by OSACA

### ISSUE THREAD

**Issue by mattgodbolt** [compiler-explorer/compiler-explorer#2319]
OSACA running on ARM code fails with:

```
Traceback (most recent call last):
  File "/opt/compiler-explorer/osaca-0.3.10/lib/python3.8/site-packages/osaca/osaca.py", line 262, in inspect
    parsed_code = parser.parse_file(code)
  File "/opt/compiler-explorer/osaca-0.3.10/lib/python3.8/site-packages/osaca/parser/base_parser.py", line 54, in parse_file
    asm_instructions.append(self.parse_line(line, i + 1 + start_line))
  File "/opt/compiler-explorer/osaca-0.3.10/lib/python3.8/site-packages/osaca/parser/parser_AArch64.py", line 286, in parse_line
    instruction_form[self.INSTRUCTION_ID] = result[self.INSTRUCTION_ID]
TypeError: 'NoneType' object is not subscriptable
```

See: https://godbolt.org/z/WxWfd9

CC @cod3monk

**Comment by partouf** [comment]
Hadn't seen this one yet. But plenty of unrecognized opcodes. Wondering if it's better to report at https://github.com/RRZE-HPC/OSACA

**Comment by mattgodbolt** [comment]
Probably, yes!

**Comment by JanLJL** [comment]
Hi Matt,

we saw this already this morning and fixed it - I just did not make a release yet.
(This is due to the directives in the beginning of the assembly. Our parser expected parameters (like `.file ` "path/to/" "file"`) to be comma separated instead of space separated)
I also added an option to specify a code range depending on the line numbers instead of markers since you cannot edit the assembly on the website. I will make a new release later today and try to make a PR here in the CE to update the version if I find the right code sections, otherwise we need @cod3monk for this ;) 

PS: Is there a way to use always OSACA's most recent release?

**Comment by mattgodbolt** [comment]
Hi @JanLJL thanks for the comment!

* Interesting re: comma separated; we're just relaying the output from the compiler directly (`clang` in this case)
* Cool! Thanks! re: code sections. I'm not sure quite what you need here though
* There's no easy way to use the most recent release: we can add support for it though. I will file an issue. We have this for some compilers and libraries, but not for tools, at least not currently. The tooling would update nightly, or on demand for us.

**Comment by JanLJL** [comment]
> * Cool! Thanks! re: code sections. I'm not sure quite what you need here though

Oh, nothing, sorry for being unspecific here, I just wanted to explain the reason why I haven't released the new version yet. But since this let us observe that the assembly file OSACA analyzes contains much more than the lines of assembly shown in the web editor, we will probably create an issue/request for enhancement for adjusting the assembly line numbers to the actual file, but this can be discussed in that issue then. 

> * There's no easy way to use the most recent release: we can add support for it though. I will file an issue. We have this for some compilers and libraries, but not for tools, at least not currently. The tooling would update nightly, or on demand for us.

Thank you! This would ease work a lot :)

**Comment by cod3monk** [comment]
fixed with #2320 
missing instructions will be included in next OSACA release, coming tonight or tomorrow.

**Comment by partouf** [comment]
This was fixed a while ago
